drivers/w1/w1_ds2482_800_channel.c

```diff
@@ -0,0 +1,180 @@
+}
+
+static int ds2482_channel_write_bit(const struct device *dev, bool bit)
+{
+	const struct ds2482_channel_config *config = dev->config;
+
+	return ds2482_84_write_bit(&config->i2c_spec, bit);
+}
+
+static int ds2482_channel_read_byte(const struct device *dev)
+{
+	const struct ds2482_channel_config *config = dev->config;
+
+	return ds2482_84_read_byte(&config->i2c_spec);
+}
+
+static int ds2482_channel_write_byte(const struct device *dev, uint8_t byte)
+{
+	const struct ds2482_channel_config *config = dev->config;
+
+	return ds2482_84_write_byte(&config->i2c_spec, byte);
+}
+
+static int ds2482_channel_configure(const struct device *dev, enum w1_settings_type type,
+				    uint32_t value)
+{
+	const struct ds2482_channel_config *config = dev->config;
+
+	uint8_t reg = config->reg_config;
+
+	switch (type) {
+	case W1_SETTING_SPEED:
+		WRITE_BIT(reg, DEVICE_1WS_pos, value);
+		break;
+	case W1_SETTING_STRONG_PULLUP:
+		WRITE_BIT(reg, DEVICE_SPU_pos, value);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return ds2482_84_write_config(&config->i2c_spec, reg);
+}
+
+static int ds2482_channel_set_channel(const struct i2c_dt_spec *spec, uint8_t channel,
+				      uint8_t channel_rb)
+{
+	int ret;
+
+	uint8_t reg = channel;
+
+	ret = ds2482_84_write(spec, CMD_CHSL, &reg);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = ds2482_84_read(spec, REG_NONE, &reg);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return (reg == channel_rb) ? 0 : -EIO;
+}
+
+static int ds2482_channel_change_bus_lock(const struct device *dev, bool lock)
+{
+	const struct ds2482_channel_config *config = dev->config;
+
+	int ret;
+
+	ret = w1_change_bus_lock(config->parent, lock);
+	if (ret < 0) {
+		LOG_ERR("Failed to acquire bus lock: %d", ret);
+		return ret;
+	}
+
+	if (!lock) {
+		return 0;
+	}
+
+	ret = ds2482_channel_set_channel(&config->i2c_spec, config->reg_channel,
+					 config->reg_channel_rb);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = ds2482_84_write_config(&config->i2c_spec, config->reg_config);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int ds2482_init(const struct device *dev)
+{
+	const struct ds2482_channel_config *config = dev->config;
+
+	if (!device_is_ready(config->parent)) {
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(config->i2c_spec.bus)) {
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static const struct w1_driver_api ds2482_driver_api = {
+	.reset_bus = ds2482_channel_reset_bus,
+	.read_bit = ds2482_channel_read_bit,
+	.write_bit = ds2482_channel_write_bit,
+	.read_byte = ds2482_channel_read_byte,
+	.write_byte = ds2482_channel_write_byte,
+	.configure = ds2482_channel_configure,
+	.change_bus_lock = ds2482_channel_change_bus_lock,
+};
+
+#define DS2482_CHANNEL_INIT(inst)                                                                  \
+	static const struct ds2482_channel_config inst_##inst##_config = {                         \
+		.w1_config.slave_count = W1_INST_SLAVE_COUNT(inst),                                \
```

Comment:
I think there is something more to consider . how is W1_INST_SLAVE_COUNT really defined ? -- i see in w1.h it is some static derivative of device tree. I need it to be independent from device tree (and large). as I have up to about 128 devices and they are randomly distributed on the 8 w1 ports of the 2482-800. The w1_net layer seems to use this in certain paths . How can I meet my objective ?